go/extractor/BUILD.bazel

```diff
@@ -0,0 +1,57 @@
+load("@rules_go//go:def.bzl", "go_library")
+load("@rules_pkg//pkg:mappings.bzl", "pkg_files")
+
+# gazelle:prefix github.com/github/codeql-go/extractor
+# gazelle:map_kind go_binary codeql_go_binary //go:rules.bzl
+
+# the immediately following `extractor` target is kept up to date by `bazel run //go:gen`, do not edit directly
+go_library(
+    name = "extractor",
+    srcs = [
+        "extractor.go",
+        "gomodextractor.go",
+        "semaphore.go",
+    ],
+    importpath = "github.com/github/codeql-go/extractor",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//go/extractor/dbscheme",
+        "//go/extractor/diagnostics",
+        "//go/extractor/srcarchive",
+        "//go/extractor/trap",
+        "//go/extractor/util",
+        "//go/extractor/vendor/golang.org/x/mod/modfile",
+        "//go/extractor/vendor/golang.org/x/tools/go/packages",
+    ],
+)
+
+# notice that these other targets are not generated
+java_library(
```

Comment:
Note to self: This picks up the right java toolchain in the semmle-code context, so we need to ensure that we actually call the build that we ship from there.
I'm also not sure how this works when building the extractor pack from the `codeql` repo: If we pick up Java 21 here, and then load this jar into the `codeql` process, it might be incompatible.
Right now, it works, as we've updated the shipped JVM to 21 already, but at times, the default Java version for bazel will be ahead of the JVM version we're shipping with CodeQL, and presumably then his breaks.
I don't know exactly where/how/with which JVM `tokenizer.jar` is executed, though.

Reply:
good points! maybe we should specify a java toolchain in this `MODULE.bazel`? Though I guess then this compilation would pick up that toolchain, and not the one from the internal repo, which might get us in a pickle with ql module updates?

Reply:
If nothing changed with bzlmod recently, setting up a Java toolchain that we'd like to use internally is a bit more complicated - I'd rather not duplicate this effort, as it'll be harder to keep in sync, and indeed get in the way of Java version upgrades via submodule bumps (although usually, there's some test changes needed in the Java extractor, so there's often submodule bumps anyways).